pkg/traceql/ast_execute.go

```diff
@@ -433,24 +433,26 @@ func (o *BinaryOperation) execute(span Span) (Static, error) {
 		case OpLessEqual:
 			return NewStaticBool(strings.Compare(lhsS, rhsS) <= 0), nil
 		case OpRegex:
-			if o.compiledExpression == nil {
-				o.compiledExpression, err = regexp.NewRegexp([]string{rhsS}, true)
+			if len(o.compiledExpressions) == 0 {
+				compiled, err := regexp.NewRegexp([]string{rhsS}, true)
```

Comment:
insight into why we made this a slice? it looks like we can still are only adding one compiled expression